src/main/java/org/opensearch/flowframework/workflow/RegisterModelGroupStep.java

```diff
@@ -142,7 +143,10 @@ public void onFailure(Exception ex) {
             String modelGroupName = (String) inputs.get(NAME_FIELD);
             String description = (String) inputs.get(DESCRIPTION_FIELD);
             List<String> backendRoles = getBackendRoles(inputs);
-            AccessMode modelAccessMode = (AccessMode) inputs.get(MODEL_ACCESS_MODE);
+            AccessMode modelAccessMode = null;
+            if (inputs.get(MODEL_ACCESS_MODE) != null) {
```

Comment:
Can we make this `inputs.containsKey()`?

Reply:
changed